### Reflections - Filip Kronstr√∂m
We had quite a lot of issues during this project, but it has at the same time been a good learning experience. One big take aways is how a majority of the paper sounded rather straight forward during the first read throughs. First when trying to recreate their steps you noticed the gaps in the methods description. It became very obvious how difficult it is to reproduce research unless this is something the authors has really aimed for. After working with this project you realize how important it is to publish code if you want your research to be reproduced. But other small things are also important, for instance we had problems downloading the data due to a majority of the ENA links being incorrect for some reason. Such small things can take quite some time to solve when doing something like this for the first time.

My background is not within biology or bioinformatics and this was the first time I worked with such a problem and such data. Just facing all the different steps and terminology in practice is very educational. Even to just decide on what project to do was a good experience. This helped me get an idea of databases for this data and also how to find the corresponding papers for studies. 

My main focus in this project was the transcriptome assembly. I think the fact it took some 20-30h to complete it for one blue mussel sample gives an idea about the complexity of this problem which is difficult to get from a lecture or by reading a book. Given that we are probably more interested in more complex species than mussels, you can definately see the continued need for increased computing resources. Apart from this I have for instance also worked on trying to find some way forward when we realized doing the transcriptome was not feasible. What we ended up doing was to merge the pair-ended reads into one file, hoping this would at least give slightly more representative data.

In different fields there is always a slightly different language used when trouble shooting and trying to find software and packages to solve problems. This project has taught me a bit about how to formulate my thoughts when googling for bioinformatics problems.

Finally, although this project has been quite frustrating, it is also great to work on projects with people from different backgrounds. For instance I feel I have learnt a lot about biology and bioinformatics from the other group members. Hopefully they feel like they have learnt a bit about technical things such as git, containers, and UNIX from me. 